packages/ckeditor5-dev-utils/lib/styles/themeimporter.js

```diff
@@ -0,0 +1,218 @@
+		options.fileToImportParent = inputFilePath;
+
+		return importFile( options );
+	}
+}
+
+// Imports a CSS file specified in the options using the postcss-import
+// plugin and appends its content to the css tree (root).
+//
+// @private
+// @param {Options} Plugin options.
+// @returns {Promise}
+function importFile( options ) {
+	const { root, result, sourceMap } = options;
+	const file = options.fileToImport;
+	const parent = options.fileToImportParent;
+	const processingOptions = {
+		from: file,
+		to: file,
+		map: sourceMap ? { inline: true } : false
+	};
+
+	if ( !fs.existsSync( file ) ) {
+		if ( options.debug ) {
+			log.info( `[ThemeImporter] Failed to find "${ chalk.yellow( file ) }".` );
+		}
+
+		return;
+	}
+
+	return postcss( options.postCssOptions )
+		.process( `@import "${ file }";`, processingOptions )
+		.then( importResult => {
+			// Merge the CSS trees.
+			root.append( importResult.root.nodes );
+
+			// Let the watcher know that the theme file should be observed too.
+			result.messages.push( {
+				file, parent,
+				type: 'dependency'
+			} );
+
+			if ( options.debug ) {
+				log.info( `[ThemeImporter] Loaded "${ chalk.green( file ) }".` );
+			}
+		} )
+		.catch( error => {
+			throw error;
+		} );
+}
+
+// For given:
+//
+// * path to the theme,
+// * and path to the CSS file processed by PostCSS,
+//
+// it returns a path to the complementary file in the theme.
+//
+// E.g., if the path to the theme is:
+//
+//		"/foo/bar/ckeditor5-theme-foo/theme/theme.css"
+//
+// and the CSS to be themed is:
+//
+//		"/baz/qux/ckeditor5-qux/theme/components/button.css"
+//
+// this helper will return:
+//
+//		"/foo/bar/ckeditor5-theme-foo/ckeditor5-qux/theme/components/button.css"
+//
+// @private
+// @param {String} themePath Path to the theme.
+// @param {String} inputFilePath Path to the CSS file which is to be themed.
+// @returns {String}
+function getThemeFilePath( themePath, inputFilePath ) {
+	// ckeditor5-theme-foo/theme/theme.css -> ckeditor5-theme-foo/theme
+	themePath = path.dirname( themePath );
+
+	// "ckeditor5-qux"
+	const packageName = getPackageName( inputFilePath );
+
+	// "components/button.css"
+	const inputFileName = inputFilePath.split( packageName + '/theme/' )[ 1 ];
+
+	// Don't load theme file for files not belonging to "ckeditor5-*/theme" folder.
+	if ( !inputFileName ) {
+		return;
+	}
+
+	// A corresponding theme file e.g. "/foo/bar/ckeditor5-theme-baz/theme/ckeditor5-qux/components/button.css".
+	return path.resolve( themePath, packageName, inputFileName );
+}
+
+// Returns a (CKEditor 5) package name the file belongs to.
+//
+// E.g., for the path to the file:
+//
+//		"/foo/ckeditor5/packages/ckeditor5-bar/baz.css"
+//
+// it outputs
+//
+//		"ckeditor5-bar"
+//
+// @private
+// @param {String} inputFilePath A path to the file.
+// @returns {String} The name of the package.
+function getPackageName( inputFilePath ) {
+	const match = inputFilePath.match( /ckeditor5-[^/]+/g );
```

Comment:
I'm not sure but this also cannot work on Windows. `[\\/]` instead of `[^/]` should help.